android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirTaskUtil.kt

```diff
@@ -108,4 +110,58 @@ constructor(@ApplicationContext val appContext: Context, val defaultRepository:
       ?.outcomeReference
       ?.lastOrNull()
       ?.extractId() == task.logicalId
+
+  suspend fun updateTaskStatuses() {
+    Timber.i("Update tasks statuses")
+
+    val tasks =
+      defaultRepository.fhirEngine.search<Task> {
+        filter(
+          Task.STATUS,
+          { value = of(Task.TaskStatus.REQUESTED.toCoding()) },
+          { value = of(Task.TaskStatus.ACCEPTED.toCoding()) },
+          { value = of(Task.TaskStatus.RECEIVED.toCoding()) },
+        )
+        filter(
+          Task.PERIOD,
+          {
+            prefix = ParamPrefixEnum.LESSTHAN_OR_EQUALS
+            value = of(DateTimeType(Date().plusDays(-1)))
+          }
+        )
+      }
+
+    Timber.i("Found ${tasks.size} tasks to be updated")
+
+    tasks.forEach { task ->
+      // expired tasks are handled by other service i.e. FhirTaskExpireWorker
+      if (task.executionStartIsBeforeOrToday() &&
+          task.status == Task.TaskStatus.REQUESTED &&
+          task.preReqConditionSatisfied()
+      ) {
```

Comment:
why not also consider the above 2 conditions part of the precondition, and add them to that function?

```suggestion
      if task.preReqConditionSatisfied() {
```

Reply:
I see. I think the conditions here are breaking the functionality for marking Tasks as ready. The conditions for updating the dependent Tasks should be separate from the parent Task.